src/dtx/dtx_rpc.c

```diff
@@ -494,11 +494,18 @@ dtx_dti_classify_one(struct ds_pool *pool, daos_handle_t tree, d_list_t *head,
 		d_iov_t			 kiov;
 		d_iov_t			 riov;
 
-		ABT_rwlock_wrlock(pool->sp_lock);
+		ABT_rwlock_rdlock(pool->sp_lock);
 		rc = pool_map_find_target(pool->sp_map,
 					  mbs->dm_tgts[i].ddt_id, &target);
+		if (rc != 1) {
+			D_ASSERT(pool_map_empty(pool->sp_map));
```

Comment:
Instead of putting such assert here, why don't we just change the pool_map_find_target() to return -DER_UNINIT when the pool map is empty?

Another question is: Why we got an uninitialized pool map here?